src/lib/y2network/sysconfig/interface_file.rb

```diff
@@ -35,22 +35,78 @@ class << self
 
         # Finds the ifcfg-* file for a given interface
         #
-        # @param name [String] Interface name
+        # @param interface [String] Interface name
         # @return [Sysconfig::InterfaceFile,nil] Sysconfig
-        def find(name)
-          return nil unless Yast::FileUtils.Exists(SYSCONFIG_NETWORK_DIR.join("ifcfg-#{name}").to_s)
-          new(name)
+        def find(interface)
+          return nil unless Yast::FileUtils.Exists(SYSCONFIG_NETWORK_DIR.join("ifcfg-#{interface}").to_s)
+          new(interface)
         end
 
-        def define_parameter(name, type = :string)
-          define_method name do
-            value = fetch(name.to_s.upcase)
+        # Defines a parameter
+        #
+        # This method adds a pair of methods to get and set the parameter's value.
+        #
+        # @param param_name [Symbol] Parameter name
+        # @param type       [Symbol] Type to be used (:string, :integer, :symbol, :ipaddr)
+        def define_parameter(param_name, type = :string)
+          params << param_name
+
+          define_method param_name do
+            return @values[param_name] if @values.key?(param_name)
+            value = fetch(param_name.to_s.upcase)
             send("value_as_#{type}", value)
           end
+
+          define_method "#{param_name}=" do |value|
+            @values[param_name] = value
+          end
+        end
+
+        # Defines an array parameter
+        #
+        # This method adds a pair of methods to get and set the parameter's values.
+        #
+        # @param param_name [Symbol] Parameter name
+        # @param limit      [Integer] Maximum array size
+        # @param type       [Symbol] Type to be used (:string, :integer, :symbol, :ipaddr)
+        def define_array_parameter(param_name, limit, type = :string)
+          define_method "#{param_name}s" do
+            return @values[param_name] if @values.key?(param_name)
+            key = param_name.to_s.upcase
+            values = [fetch(key)]
+            values += Array.new(limit) do |idx|
+              value = fetch("#{key}_#{idx}")
+              next if value.nil?
+              send("value_as_#{type}", value)
+            end
+            values.compact
```

Comment:
I am not sure if it is problem, but this compact basically change file. 
if there is KEY_1, KEY_3, then after write there will be KEY_1, KEY_2.

Reply:
Good point. I do not think it is a problem as we are not going to write the array unless we have explicitly modified it. But it is true that it does not reflect the real content. I will fix it.

Reply:
Thinking about that, we could solve this as part of [the upcoming support for multiple IPs](https://trello.com/c/0lBPS32M/997-network-ng-start-with-a-data-model-which-permits-to-use-different-network-connection-profiles).